lib/api/core/models/repositories/roleRepository.js

```diff
@@ -118,6 +118,53 @@ module.exports = function (kuzzle) {
   };
 
   /**
+   * Get from database the document that represent the role given in parameter
+   * @param {Role} role
+   * @returns {Promise}
+   */
+  RoleRepository.prototype.loadRole = function (role) {
+    var deferred = q.defer();
+
+    if (!role._id) {
+      deferred.reject(new BadRequestError('Missing role id'));
+      return deferred.promise;
+    }
+
+    if (this.roles[role._id]) {
+      deferred.resolve(this.roles[role._id]);
+    }
+    else {
+      return this.loadOneFromDatabase(role._id);
```

Comment:
We can't use `load` function from repository. There is an extract of this function

```
if (this.cacheEngine === null) {
    return this.loadOneFromDatabase(id);
  }

  this.loadFromCache(id, opts).then()
```

And for RoleRepository the cache is undefined. But like you can see line 133 in roleRepository, I check if the role is not in memory before load it from database.